lib/testscript_engine/testscript_runnable.rb

```diff
@@ -176,14 +177,17 @@ def load_fixtures
   end
 
   def load_profiles
+    puts ("      Loading profiles...")
     script.profile.each do |profile|
       next warning(:no_static_profile_id) unless profile.id
       next warning(:no_static_profile_reference) unless profile.reference
 
       profile_server = FHIR::Client.new("")
       response = profile_server.send(:get, profile.reference, { 'Content-Type' => 'json' })
-      next if response.response[:code].to_s.starts_with?('2')
-
+      if response.response[:code].to_s != "200"
+        puts ("      Failed to load profile '#{profile.id}' from '#{profile.reference}': Response code #{response.response[:code]}")
+        next 
+      end
       profiles[profile.id] = FHIR.from_contents(response.response[:body].to_s)
```

Comment:
When the external validator is in use, then this load isn't relevant to profile validation.  Instead, you have to load into the validator, or make sure it is already present. The [list supported profiles](https://github.com/inferno-framework/fhir-validator-wrapper/blob/master/rest-api.md#list-supported-profiles) route can be used to check if it is already present and the [load custom profile](https://github.com/inferno-framework/fhir-validator-wrapper/blob/master/rest-api.md#load-a-custom-profile) route can be used to load a structure definition.

If the URL doesn't exist and an external validator is in use, checking whether the profile is already present and succeeding if so would allow a kludgy, but possible way to support under-development profiles: fire up a validator instance, load your profile on it, then run against that validator instance.

Reply:
Good comments! I will work on these.

Reply:
Done : https://github.com/fhir-crucible/testscript-engine/pull/70/commits/f09ff32dcc1c2c817d352816cfeeb87d2d8e869c